tests/chart_tests/test_byo_sa.py

```diff
@@ -0,0 +1,36 @@
+import pytest
+
+from tests import supported_k8s_versions
+from tests.chart_tests.helm_template_generator import render_chart
+
+
+@pytest.mark.parametrize(
+    "kube_version",
+    supported_k8s_versions,
+)
+class TestServiceAccounts:
+    def test_SAs_created(self, kube_version):
+        # Render the chart with rbacEnabled set to False
+        docs = render_chart(kube_version=kube_version, values={"global": {"rbacEnabled": False}})
+
+        # Check that no ServiceAccount resources are created
+        """
+        service_accounts = [
+            doc["metadata"]["name"] for doc in docs if isinstance(doc, dict) and doc.get("kind") == "ServiceAccount"
+        ]
+        assert len(service_accounts) == 0, "No ServiceAccounts should be created when rbacEnabled is False"
+        """
+        # Check that the Deployment or StatefulSet is using the default ServiceAccount
+        for doc in docs:
+            if isinstance(doc, dict) and doc.get("kind") in ["Deployment", "StatefulSet"]:
```

Comment:
No need to test if it's a dict. It is. If it's not we need to know that, so we should fail.

Reply:
done